packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visit-summary.component.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import Tabs from 'carbon-components-react/es/components/Tabs';
+import Tab from 'carbon-components-react/es/components/Tab';
+import dayjs from 'dayjs';
+import { Order, Encounter, Note, Observation, OrderItem, TestItem } from '../visit.resource';
+import styles from '../visit-detail-overview.scss';
+import MedicationSummary from './medications-summary.component';
+import NotesSummary from './notes-summary.component';
+import TestsSummary from './tests-summary.component';
+import { OpenmrsResource } from '@openmrs/esm-framework';
+
+function formatTime(date) {
+  return dayjs(date).format('hh:mm');
+}
+
+interface DiagnosisItem {
+  diagnosis: string;
+  order: string;
+}
+
+interface VisitSummaryProps {
+  encounters: Array<Encounter | OpenmrsResource>;
+}
+
+const VisitSummary: React.FC<VisitSummaryProps> = ({ encounters }) => {
+  const { t } = useTranslation();
+  const [tabSelected, setSelectedTab] = useState(0);
+
+  const [diagnoses, notes, tests, medications]: [Array<DiagnosisItem>, Array<Note>, Array<TestItem>, Array<OrderItem>] =
+    useMemo(() => {
+      // Medication Tab
+      let medications: Array<OrderItem> = [];
+      // Diagnoses in a Visit
+      let diagnoses: Array<DiagnosisItem> = [];
+      // Notes Tab
+      let notes: Array<Note> = [];
+      // Tests Tabs
+      let tests: Array<TestItem> = [];
+
+      // Iterating through every Encounter
+      encounters.forEach((enc: Encounter) => {
+        // Orders of every encounter put in a single array.
+        medications = medications.concat(
```

Comment:
Hm why not make `medications` a const and instead of re-assigning use `medications.push(...otherArray)`?

Reply:
If you see line 31, in the end, it is being assigned to a constant only. Do you want me to go for this?